doc/users_guide.rst

```diff
@@ -0,0 +1,306 @@
+
+Once these two components are available on your machine, issue these commands::
+
+  git clone https://github.com/ARM-software/lisa.git -b next
+  cd lisa
+  vagrant up
+
+This last command builds and executes the VM according to the description provided
+by the Vagrant file available in the root folder of the LISA source tree.
+
+Once the VM installation is complete, you can access that VM with::
+
+  vagrant ssh
+
+LISA shell
+==========
+
+You should now have all of the required dependencies installed. From now on, you
+can use the LISA shell, which provides a convenient set of commands for easy
+access to many LISA related functions (done automatically by Vagrant)::
+
+  source init_env
+
+Run ``lisa-help`` to see an overview of the provided LISA commands.
+
+Notebooks
+=========
+
+The LISA shell can simplify starting an Jupyter notebook server::
+
+  [LISAShell lisa] \> lisa-jupyter start
+
+  Starting Jupyter Notebooks...
+
+  Notebook server configuration:
+    URL        :  http://127.0.0.1:8888/?token=b34F8D0e457BDa570C4A6D7AF113CB45d9CcAF44Aa7Cf400
+    Folder     :  /data/work/lisa/ipynb
+    Logfile    :  /data/work/lisa/ipynb/server.log
+    PYTHONPATH :
+	  /data/work/lisa/modules_root/
+
+
+  Notebook server task: [4] 30177
+
+Note that the ``lisa-jupyter`` command allows you to specify interface and
+port in case you have several network interfaces on your host::
+
+  lisa-jupyter start [interface [port]]
+
+The URL of the main folder served by the server is printed on the screen.
+By default it is http://127.0.0.1:8888/. Once the server is started you can
+have a look at the provided tutorial notebooks are accessible by following this
+`link <http://127.0.0.1:8888/notebooks/examples/typical_experiment.ipynb>`__.
+This initial tutorial can be seen (but not executed) also on `github
+<https://github.com/ARM-software/lisa/blob/next/ipynb/examples/typical_experiment.ipynb>`__.
+
+
+Transitioning from LISA legacy
+==============================
+
+A big refactoring effort was started in mid 2018, which produced a lot of
+(much needed) changes. If you are used to using LISA before the refactoring came
+into place, this guide is for you.
+
+Project structure
++++++++++++++++++
+
+* ``lisa/libs/utils`` is now ``lisa/lisa/``. ``lisa/libs/wlgen`` has also been
+  moved to that location.
+* devlib, trappy and bart are now under ``lisa/external/``. Git subtrees are
+  now used instead of submodules.
+* ``lisa/tests/lisa`` is now ``lisa/tests``
+* ``lisa/tests/eas`` is now ``lisa/lisa/tests``
+
+Imports
++++++++
+
+LISA legacy used relative imports with a bit of dark magic to hold everything
+together. Say you want to import LISA the :class:`lisa.trace.Trace` class found
+in ``lisa/trace.py``, previously you would do it like so::
+
+  from trace import Trace
+
+However, relative imports are dangerous - did you know :class:`trace.Trace` exists
+in Python's standard library? This means that the above import would import done
```

Comment:
this sentence needs some fixes to be parseable :)
Also, `from trace import Trace` will still give the stdlib `Trace`, even in LISA environment, since that is not a relative import. Imports exist in 3 flavors:

1) explicit: full name from top-level package, e.g. `import lisa.foo.bar`
2) relative: equivalent to full name, but less obvious and potentially confusing error messages when they are broken, e.g. 'from . import bar' (assuming the current module is at the same level as `lisa.foo`)
3) implicit relative import: Python2-only evil, which introduces ambiguity, e.g. `import bar`. It is not obvious that `bar` is actually `lisa.foo.bar`, and it can shadow an stdlib module named `bar` (just like `trace.Trace` that was shadowed, as it already existed in Python 2). That is very nasty as it becomes impossible to import some modules from the stdlib, which is quite broken.

To sum up:
* really dangerous imports (implicit relative) are not possible in Python 3
* explicit relative imports are still not recommended, for code clarity, grepping and sensible error messages when something goes wrong.
* absolute imports are the way to go, unless there is a good justification the other types should be banned from the code base.